contracts/ConferenceTicket.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.5.11;
+
+import './Conference.sol';
+import './GroupAdmin.sol';
+import './IConferenceTicket.sol';
+import {Utils} from './Utils.sol';
+import 'openzeppelin-solidity/contracts/token/ERC721/ERC721.sol';
+
+contract ConferenceTicket is IConferenceTicket, ERC721, GroupAdmin {
+    // Base Token URI
+    string public baseTokenURI;
+    address public conferenceAddress;
+    Conference public conference;
+
+    // Optional mapping for token URIs
+    mapping(uint256 => string) private _tokenURIs;
+
+    modifier onlyAdminOrContract() {
+        require(isAdmin(msg.sender) || msg.sender == conferenceAddress, 'must be admin or contract');
+        _;
+    }
+
+    constructor(string memory _baseTokenUri) public {
+        baseTokenURI = _baseTokenUri;
+    }
+
+    function setBaseTokenURI(string calldata uri) external onlyAdmin {
+        baseTokenURI = uri;
+    }
+    
+    function setConferenceAddress(address _address) external onlyAdmin {
```

Comment:
If `Conference` instantiate the `ConferenceTicket` or you don't separate it into two contracts, is there a need to change the conference address?

Reply:
Maybe this part is not clear.

`Each event (Conference) has its own NFT contract` means each conference has its own NFT contract rather than one contract that represents multiple events, not about NFT contract and event contract being seperated

Reply:
Changing the conference address would not be needed if we could instantiate the `ConferenceTicket` inside the constructor of `AbstractConference`. But since we are constrained by the contract size limit, it makes an elegant approach in my opinion.

Reply:
What is the block gas limit you set on your local dev environment? Polygon mainnet seems to have 20M. Did it even hit the size?

Reply:
The default value `12450000` as I did not change the config settings. Running with `hardhat-contract-sizer` returns:

![grafik](https://user-images.githubusercontent.com/4712166/127775099-90e789d3-858c-4d19-bdd1-14c3da3dc42d.png)

Seems to be fine for Polygon mainnet then. How would you like to proceed?

Reply:
We currently support xDAI and Polygon. [xDAI block size is 17M and Polygon is 20M](https://www.xdaichain.com/about-xdai/news-and-information/comparisons/matic ).
 If the contract fits in less than 17M, would like to be one contract as it will be simpler.  cc @dantop114

Reply:
Note: Please also try to actually deploy to Polygon mainnet to make sure it works.

Reply:
Can you send me some funds to deploy on Polygon mainnet? 0x61d64AfBbD3b5CC2D0554A8a92aa5C7540501E7c

Reply:
Sent you 3 matic